# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../sane-backends/types.h.patch.musl
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- sane-backends-1.0.25/include/sane/sanei_udp.h.vanilla	2019-08-13 14:24:34.959655184 +0000
+++ sane-backends-1.0.25/include/sane/sanei_udp.h	2019-08-13 14:25:54.531652862 +0000
@@ -18,6 +18,9 @@
 #ifndef sanei_udp_h
 #define sanei_udp_h
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
 #include <sane/sane.h>
 
 #ifdef HAVE_WINSOCK2_H
--- sane-backends-1.0.25/backend/epsonds-io.c.vanilla	2019-08-13 14:26:33.419651728 +0000
+++ sane-backends-1.0.25/backend/epsonds-io.c	2019-08-13 14:26:43.575651431 +0000
@@ -17,6 +17,9 @@
 #include <ctype.h>
 #include <unistd.h>     /* sleep */
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
 #include "epsonds.h"
 #include "epsonds-io.h"
 
--- sane-backends-1.0.25/backend/hp5590_low.c.vanilla	2019-08-13 14:26:58.911650984 +0000
+++ sane-backends-1.0.25/backend/hp5590_low.c	2019-08-13 14:27:09.583650672 +0000
@@ -51,6 +51,10 @@
 # include <netinet/in.h>
 #endif /* HAVE_NETINET_IN_H */
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
+
 #include "byteorder.h"
 
 #include "../include/sane/sanei_debug.h"
--- sane-backends-1.0.25/backend/hp5400_sanei.c.vanilla	2019-08-13 14:27:52.391649423 +0000
+++ sane-backends-1.0.25/backend/hp5400_sanei.c	2019-08-13 14:28:06.103649023 +0000
@@ -58,6 +58,9 @@
 #include <stdio.h>
 #include "../include/sane/sanei_usb.h"
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
 #define CMD_INITBULK1   0x0087	/* send 0x14 */
 #define CMD_INITBULK2   0x0083	/* send 0x24 */
 #define CMD_INITBULK3   0x0082	/* transfer length 0xf000 */
--- sane-backends-1.0.25/backend/kvs20xx_cmd.h.vanilla	2019-08-13 14:28:24.147648496 +0000
+++ sane-backends-1.0.25/backend/kvs20xx_cmd.h	2019-08-13 14:28:32.651648248 +0000
@@ -9,6 +9,9 @@
    Panasonic KV-S20xx USB-SCSI scanners.
 */
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
 #define COMMAND_BLOCK	1
 #define DATA_BLOCK	2
 #define RESPONSE_BLOCK	3
--- sane-backends-1.0.25/backend/kvs40xx.h.vanilla	2019-08-13 14:29:02.415647380 +0000
+++ sane-backends-1.0.25/backend/kvs40xx.h	2019-08-13 14:29:09.299647179 +0000
@@ -11,6 +11,10 @@
 #include "../include/sane/config.h"
 #include <semaphore.h>
 
+#define _GNU_SOURCE
+#include <sys/types.h>
+
+
 #undef  BACKEND_NAME
 #define BACKEND_NAME kvs40xx
 
